pkg/aggregator/context_resolver.go

```diff
@@ -97,11 +128,56 @@ func (cr *ContextResolver) expireContexts(expireTimestamp float64) []ckey.Contex
 		}
 	}
 
+	cr.resolver.removeKeys(expiredContextKeys)
+
 	// Delete expired context keys
 	for _, expiredContextKey := range expiredContextKeys {
-		delete(cr.contextsByKey, expiredContextKey)
 		delete(cr.lastSeenByKey, expiredContextKey)
 	}
 
 	return expiredContextKeys
 }
+
+// countBasedContextResolver allows tracking and expiring contexts based on the number
+// of calls of `expireContexts`.
+type countBasedContextResolver struct {
+	resolver               *contextResolver
+	keyByContextCountIndex map[ckey.ContextKey]int64
```

Comment:
Shouldn't this be `contextCountIndexByKey` instead? (cf `lastSeenByKey` in the timestamp based resolver).

(and this is just a personal preference, but there are too many nouns stringed together here; could this be just `indexByKey`?)